lib/AppRoutes.js

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useRef } from 'react';
+import { Route, withRouter } from 'react-router-dom';
+
+import MainBuilder from './builder/components/Main';
+import Editor from './builder/components/Editor';
+import MainViewer from './viewer/components/Main';
+import Viewer from './viewer/components/Editor';
+
+const AppRoutes = props => {
+  const {
+    config,
+    location: { pathname: currentPathname },
+    onRouteChange
+  } = props;
+  const prevLocationRef = useRef();
+  useEffect(() => {
+    prevLocationRef.current = currentPathname;
+  }, [props]);
+  const prevPathname = prevLocationRef.current;
+
+  prevPathname &&
+    onRouteChange &&
+    prevPathname !== currentPathname &&
+    onRouteChange();
+
+  return (
```

Comment:
Why not just 

`
<Route path="/" component={() => {
   onRouteChange();
   return null;
}}> />
`

Reply:
True :)